research/conditional/utils/layer_manager.py

```diff
@@ -44,19 +44,23 @@ def _register_layers(self, model):
                 self._layers.append((block_name, layer))
 
     def prepare_for_logging(self, step):
-        if (
-            step % self.logging_interval_light == 0
-            or step % self.logging_interval_heavy == 0
+        if step % self.logging_interval_light == 0 or (
+            self.logging_interval_heavy > 0
+            and (step % self.logging_interval_heavy == 0)
         ):
             for block_name, layer in self._layers:
                 if hasattr(layer, "prepare_for_logging"):
                     layer.prepare_for_logging()
 
     def log(self, step):
         verbosity_levels = []
-        if step % self.logging_interval_heavy == 0:
+        if (self.logging_interval_heavy > 0) and (
+            step % self.logging_interval_heavy == 0
+        ):
```

Comment:
no need for parentheses except the outermost ones

Reply:
changed